jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -237,20 +241,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
             if (!included && !passConditionalHeaders(request,response,content))
                 return;
                 
-            // Gzip?
-            HttpContent gzip_content = gzippable?content.getGzipContent():null;
-            if (gzip_content!=null)
+            // Precompressed variant available?
+            Map<CompressedContentFormat,? extends HttpContent> precompressedContents = checkPrecompressedVariants?content.getPrecompressedContents():null;
+            if (precompressedContents!=null)
             {
                 // Tell caches that response may vary by accept-encoding
                 response.addHeader(HttpHeader.VARY.asString(),HttpHeader.ACCEPT_ENCODING.asString());
                 
-                // Does the client accept gzip?
                 String accept=request.getHeader(HttpHeader.ACCEPT_ENCODING.asString());
-                if (accept!=null && accept.indexOf("gzip")>=0)
+                CompressedContentFormat precompressedContentEncoding = getBestPrecompressedContent(accept, precompressedContents.keySet());
```

Comment:
My original code did choose the best available resource in the browser given quality order, and with fewer memory objects created. But I now modified it to use the QuotedQualiytCSV which makes the code much more clean and readable.